e2e/shared-mode-smoke-test.sh

```
@@ -0,0 +1,398 @@
+cleanup() {
+  local rc=$?
+  [ -n "$SERVER_PID" ] && kill -0 "$SERVER_PID" >/dev/null 2>&1 && {
+    kill "$SERVER_PID" >/dev/null 2>&1 || true
+    wait "$SERVER_PID" 2>/dev/null || true
+  }
+  if [ "$KEEP_DB" != "1" ]; then
+    [ -n "$MYSQL_CONTAINER" ] && "$DB_RUNTIME" rm -f "$MYSQL_CONTAINER" >/dev/null 2>&1 || true
+    [ -n "$TIDB_CONTAINER" ] && "$DB_RUNTIME" rm -f "$TIDB_CONTAINER" >/dev/null 2>&1 || true
+    [ -n "$NETWORK" ] && "$DB_RUNTIME" network rm "$NETWORK" >/dev/null 2>&1 || true
+  fi
+  [ "$rc" -eq 0 ] && rm -rf "$TMP_DIR" || log "failed; artifacts kept at $TMP_DIR"
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)